include/engine/plugins/trip.hpp

```diff
@@ -291,13 +287,14 @@ template <class DataFacadeT> class RoundTripPlugin final : public BasePlugin
         {
             const auto component_size = scc.range[k + 1] - scc.range[k];
 
-            BOOST_ASSERT_MSG(component_size >= 0, "invalid component size");
+            BOOST_ASSERT_MSG(component_size > 0, "invalid component size");
+
+            std::vector<NodeID> scc_route;
```

Comment:
as this is only used to transfer it into route_result, wouldn't it be more efficient to directly emplace_back the different choices (e.g. `route_results.emplace_back(trip::BruteFroceTrip())`) instead of first constructing them into scc_route and then moving them out?